src/sentry/static/sentry/app/views/onboarding/projectSetup/eventWaiter.tsx

```diff
@@ -14,30 +14,32 @@ const recordAnalyticsFirstEvent = ({organization, project}) =>
   });
 
 /**
- * This is a render prop component that can be used to wait for the first event
- * of a project to be received via polling.
- *
- * When an event is received the {firstIssue} will be passed to the child.
  * Should no issue object be available (the first issue has expired) then it
- * will simply be boolean true.
- *
- * Otherwise this value will be null before the event is received.
+ * will simply be boolean true. When no event has been recieved this will be
+ * null. Otherwise it will be the group
  */
-class EventWaiter extends React.Component {
-  static propTypes = {
-    api: PropTypes.object,
-    organization: SentryTypes.Organization,
-    project: SentryTypes.Project,
-    disabled: PropTypes.bool,
-    children: PropTypes.func,
-    pollInterval: PropTypes.number,
-  };
+type FirstIssue = null | true | Group;
```

Comment:
@EvanPurkhiser Really wondering the logic of letting `firstIssue` be `true` or a `Group`.

Reply:
It's becuase you can have sent your first issue, but because of event retention it will drop off after X days, so the while you may have sent that first event, the group may no longer exist.

The comment above explains it.